test/e2e/storage/vsphere/nodemapper.go

```diff
@@ -123,6 +139,144 @@ func (nm *NodeMapper) GenerateNodeMap(vSphereInstances map[string]*VSphere, node
 	return nil
 }
 
+// Establish rest connection to retrieve tag manager stub
+func withTagsClient(ctx context.Context, connection *VSphere, f func(c *rest.Client) error) error {
+	c := rest.NewClient(connection.Client.Client)
+	user := neturl.UserPassword(connection.Config.Username, connection.Config.Password)
+	if err := c.Login(ctx, user); err != nil {
+		return err
+	}
+	defer c.Logout(ctx)
+	return f(c)
+}
+
+// Iterates over each node and retrieves the zones in which they are placed
+func retrieveZoneInformationForNode(nodeName string, connection *VSphere, hostSystemRef types.ManagedObjectReference) []string {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	var zones []string
+	pc := connection.Client.ServiceContent.PropertyCollector
+	withTagsClient(ctx, connection, func(c *rest.Client) error {
+		client := tags.NewManager(c)
+		// Example result: ["Host", "Cluster", "Datacenter"]
+		ancestors, err := mo.Ancestors(ctx, connection.Client, pc, hostSystemRef)
+		if err != nil {
+			return err
+		}
+
+		var validAncestors []mo.ManagedEntity
+		// Filter out only Datacenter, ClusterComputeResource and HostSystem type objects. These objects will be
```

Comment:
Why filter here?  The only type I can think of that this would exclude is a `Folder`, which should be a valid object to tag.  Or is there another type that could be excluded here?

Reply:
The type currently being excluded is Folder. As per the current implementation for zone support in PR #72687, zones are discovered only from Host, Cluster and Datacenter type objects.